src/main/java/net/neoforged/neoforge/event/village/VillagerTradesEvent.java

```diff
@@ -28,10 +29,20 @@
 public class VillagerTradesEvent extends Event {
     protected Int2ObjectMap<List<ItemListing>> trades;
     protected VillagerProfession type;
+    private final RegistryAccess registryAccess;
 
+    /**
+     * @deprecated Use {@link #VillagerTradesEvent(Int2ObjectMap, VillagerProfession, RegistryAccess)} instead
+     */
+    @Deprecated(forRemoval = true, since = "1.21")
     public VillagerTradesEvent(Int2ObjectMap<List<ItemListing>> trades, VillagerProfession type) {
+        this(trades, type, RegistryAccess.EMPTY);
+    }
+
+    public VillagerTradesEvent(Int2ObjectMap<List<ItemListing>> trades, VillagerProfession type, RegistryAccess registryAccess) {
```

Comment:
Should we perhaps mark this with `@ApiStatus.Internal`? I can't think of a reason why a mod would construct this event, since NeoForge does it for all villager professions already.

Reply:
So we deprecate for removal the original constructor and mark the new one as internal, that makes sense to me as long as devs can still extend the method if they want right?

Reply:
What do you mean by "devs can still extend the method if they want"? If we intend to allow mods to fire this event, then we'd have to not mark this with the internal annotation.

Reply:
You can ignore that sorry, I had just gotten in from work and it was like 2am, I've marked it as internal in a commit like you suggested
I honestly don't know what I meant